server/artifacts/artifact_server.go

```diff
@@ -137,8 +137,7 @@ func (a *ArtifactServer) getArtifact(ctx context.Context, wf *wfv1.Workflow, nod
 	if err != nil {
 		return nil, err
 	}
-
-	tmp, err := ioutil.TempFile(".", "artifact")
+	tmp, err := ioutil.TempFile("/tmp", "artifact")
```

Comment:
Will it be automatically deleted? Or do we need to delete it? 
https://golang.org/pkg/io/ioutil/#TempFile

Reply:
yes - we should fix that bug too

Reply:
this change broke the output artifact downloads for us, we upgraded from 2.11.5 to 2.11.7, and we began to get errors like this:

`open /tmp/artifact058606664: no such file or directory`

reverting the upgrade fixed the downloads.

Reply:
@ekesken - likely using runAsNonRoot? You'll need to mount /tmp as an emptyDir volume on argo-server pod so you can write to it. This path should probably be configurable as an argument.

Reply:
I see that the fix in helm chart is already merged here: https://github.com/argoproj/argo-helm/commit/9939da51cfabe465ccb730a88976ce2a6b26112e

It was surprising to see a non-backward compatible change in a patch version change, that's why I wanted to point it here.